lib/Doctrine/Common/Annotations/CachedReader.php

```diff
@@ -51,16 +53,24 @@ final class CachedReader implements Reader
     private $loadedAnnotations = [];
 
     /**
-     * Constructor.
-     *
-     * @param Reader $reader
-     * @param Cache  $cache
-     * @param bool   $debug
+     * @param Cache|CacheInterface $cache
+     * @param bool                 $debug
      */
-    public function __construct(Reader $reader, Cache $cache, $debug = false)
+    public function __construct(Reader $reader, $cache, $debug = false)
     {
+        if ($cache instanceof Cache) {
```

Comment:
I'd probably rather deprecate this ctor, and instead add a named ctor for the new cache.

Reply:
But the named ctor would have to call that deprecated ctor, wouldn't it? Or is there a way around that I'm not aware of?

Reply:
Correct: we can then make the ctor `private` later on

Reply:
Sorry, I don't see how this is going to work… on one hand, the named ctor should call the deprecated ctor, on the other hand, the type declaration will make that impossible since it will prevent feeding a PSR-16 cache instance to the deprecated constructor.

Reply:
Instead of adding a `trigger_error`, we deprecate the public ctor, and add a new one. The type hint has to go regardless

Reply:
I would suggest throwing a deprecation message if the constructor is called with a doctrine/cache instance instead of PSR-16. The general `@deprecated` message on the constructor should stay there as well. This would appropriately trigger the deprecation for people using the Symfony PHPUnit Bridge as well as people using tools like phpstan-deprecation-rules.

Reply:
That's what I originally did. Restored :)